integration/tests/epochs/cohort2/epoch_recover_from_efm_test.go

```diff
@@ -290,3 +291,120 @@ func (s *RecoverEpochSuite) TestRecoverEpochNodeEjected() {
+// 4. Eject consensus node from the Random Beacon committee by modifying the snapshot before generating the recover epoch transaction args.
+// 5. Submit recover epoch transaction.
+// 6. Ensure expected EpochRecover event is emitted.
+// 7. Ensure the network transitions into the recovery epoch and finalizes the first view of the recovery epoch.
+func (s *RecoverEpochSuite) TestRecoverEpochEjectNodeDifferentDKG() {
+	// 1. Manually trigger EFM
+
+	// pause the collection node to trigger EFM by failing DKG
+	ln := s.GetContainersByRole(flow.RoleCollection)[0]
+	require.NoError(s.T(), ln.Pause())
+	s.AwaitFinalizedView(s.Ctx, s.GetDKGEndView(), 2*time.Minute, 500*time.Millisecond)
+	// start the paused collection node now that we are in EFM
+	require.NoError(s.T(), ln.Start())
+
+	// get final view from the latest snapshot
+	epoch1FinalView, err := s.Net.BootstrapSnapshot.Epochs().Current().FinalView()
+	require.NoError(s.T(), err)
+
+	// Wait for at least the first view past the current epoch's original FinalView to be finalized.
+	s.TimedLogf("waiting for epoch transition (finalized view %d)", epoch1FinalView+1)
+	s.AwaitFinalizedView(s.Ctx, epoch1FinalView+1, 2*time.Minute, 500*time.Millisecond)
+	s.TimedLogf("observed finalized view %d", epoch1FinalView+1)
+
+	// assert that we are in EFM
+	snapshot, err := s.Client.GetLatestProtocolSnapshot(s.Ctx)
+	require.NoError(s.T(), err)
+	epochPhase, err := snapshot.EpochPhase()
+	require.NoError(s.T(), err)
+	require.Equal(s.T(), flow.EpochPhaseFallback, epochPhase, "network must enter EFM by this point")
+
+	// 2. Generate transaction arguments for epoch recover transaction.
+	collectionClusters := s.NumOfCollectionClusters
+	recoveryEpochCounter := uint64(1)
+
+	// read internal node info from one of the consensus nodes
+	internalNodePrivInfoDir, nodeConfigJson := s.getNodeInfoDirs(flow.RoleConsensus)
+	internalNodes, err := common.ReadFullInternalNodeInfos(unittest.Logger(), internalNodePrivInfoDir, nodeConfigJson)
+	require.NoError(s.T(), err)
+	// 3. Eject consensus node by modifying the snapshot before generating the recover epoch transaction args.
+	// By ejecting a node from the consensus committee but keeping it in the Random Beacon committee, we ensure that the there is a node
+	// which is not part of the consensus committee but is part of the Random Beacon committee.
+	currentIdentityTable := snapshot.Encodable().SealingSegment.LatestProtocolStateEntry().EpochEntry.CurrentEpochIdentityTable
+	ejectedIdentity := currentIdentityTable.Filter(filter.HasRole[flow.Identity](flow.RoleConsensus))[0]
+	ejectedIdentity.EpochParticipationStatus = flow.EpochParticipationStatusEjected
+
+	// 4. Modify DKG data by removing the last node of the consensus committee from DKG committee. We can do this
+	// by altering the DKG index map and DKG public key shares.
+	// This way we ensure that consensus committee has a node which is not part of the Random Beacon committee.
```

Comment:
I was very confused why we could just remove a node from the DKG and the resulting setup would still work. In general, threshold key sets do not allow that. We are kind of using a hack here, which I think is fine for the test. Nevertheless, we should document it accordingly. 

```suggestion
	// 4. Modify DKG data by removing the last node of the consensus committee from DKG committee. This way we ensure that consensus
	// committee has a node which is not part of the Random Beacon committee. For threshold committees of *even size*, we can remove a
	// single node without changing the threshold (see [ref. 1] for details). In other words, we can just pretend that there was originally
	// one node less in the DKG, while the same number of signatures (threshold +1) are sufficient to construct a group signature.
	//
	// [ref. 1] function `RandomBeaconThreshold` for computing the threshold in package module/signature; note
	//          that for reconstructing the group sig, _strictly more_ than `threshold` sig shares are required.
```